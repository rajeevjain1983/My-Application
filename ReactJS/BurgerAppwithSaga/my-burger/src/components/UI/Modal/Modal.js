import React, {Component} from 'react';
import Classes from './Modal.css';
import Aux from '../../../hoc/Aux/Aux';
import BackDrop from '../../UI/Backdrop/Backdrop';

class Modal extends Component{
    shouldComponentUpdate(nextProps, nextState){
       return nextProps.show!==this.props.show || nextProps.children!==this.props.children;
    }
        render(){
            console.log('model Render');
            return(
                <Aux>
                <BackDrop clicked={this.props.modalClosed} show={this.props.show}></BackDrop>
                <div className={Classes.Modal} 
                style={{
                    transform: this.props.show ? 'translateY(0)' : 'translateY(-100vh)',
                    opacity: this.props.show ? '1' : '0'
                }}>
                    {this.props.children}
                </div>
        </Aux>
        )
        }
}
export default Modal;