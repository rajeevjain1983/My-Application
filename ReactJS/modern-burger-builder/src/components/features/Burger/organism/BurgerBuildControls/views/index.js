export { default } from "./BurgerBuildControls.view";
