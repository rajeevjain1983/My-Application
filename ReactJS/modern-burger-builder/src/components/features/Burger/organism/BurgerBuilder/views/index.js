export { default } from "./BurgerBuilder.view";
