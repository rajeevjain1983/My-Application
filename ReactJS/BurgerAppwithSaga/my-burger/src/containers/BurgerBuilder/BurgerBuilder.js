import React, {Component} from 'react';
import Aux from '../../hoc/Aux/Aux';
import Burger from '../../components/Burger/Burger';
import BuildControls from '../../components/Burger/BuildControls/BuildControls';
import Modal from '../../components/UI/Modal/Modal';
import OrderSummery from '../../components/Burger/OrderSummery/OrderSummery';
import axios from '../../axios';
import Spinner from '../../components/UI/Spinner/Spinner';
import WithErrorHandler from '../../hoc/WithErrorHandler/WithErrorHander';
import {connect} from 'react-redux';
import {addIngredients,removeIngredients, initIngredients} from '../../store/actions/index';


class BurgerBuilder extends Component{

    
    state={
        purchasing:false,
        loading:false,
    }

    orderNowhandler =()=>{
        this.setState({purchasing:true});
    }
    updatePurchable=(ingredients)=>{
        const sum= Object.keys(ingredients).map(igkey=>{
            return ingredients[igkey];
        })
        .reduce((sum,el)=>{
            return sum+el;
        },0)
        
        return sum>0;
    }
    // addIngredientsHandler=(type)=>{
    //     const oldCount= this.props.ings[type];
    //     const updatedCount= oldCount+1;
    //     const updatedIngredients={...this.state.ingredients};
    //     updatedIngredients[type]=updatedCount;
    //     const oldPrice=this.state.totalPrice;
    //     const updatedPrice=oldPrice+INGREDIENTS_PRICE[type];
    //     this.setState({totalPrice:updatedPrice,ingredients:updatedIngredients})
    //     this.updatePurchable(updatedIngredients);
    // }

    // removeIngredientsHandler=(type)=>{
    //     const oldCount= this.state.ingredients[type];

    //     if(oldCount<=0)
    //         return;
    //     const updatedCount= oldCount-1;
    //     const updatedIngredients={...this.state.ingredients};
    //     updatedIngredients[type]=updatedCount;
    //     const oldPrice=this.state.totalPrice;
    //     const updatedPrice=oldPrice-INGREDIENTS_PRICE[type];
    //     this.setState({totalPrice:updatedPrice,ingredients:updatedIngredients})
    //     this.updatePurchable(updatedIngredients);
    // }
    ModalClosedHandler = ()=>{
        this.setState({purchasing:false});
    }
    ContinueOrderHandler=()=>{
        const quryParams =[];
        for(let i in this.props.ings){
            quryParams.push(encodeURIComponent(i) + '='+ encodeURIComponent(this.props.ings[i]));
        }
        quryParams.push('price='+this.state.totalPrice);
        const queryString = quryParams.join('&');
        // console.log(queryString);
       this.props.history.push({
           pathname:'/checkout',
           search: '?'+queryString
        });
        
    }
    componentDidMount(){ 
        this.props.loadIngredients();
    }
    render(){
        const disabledInfo={...this.props.ings};
        for(let key in disabledInfo){
            disabledInfo[key]=disabledInfo[key]<=0;
        }

        let burger=<Spinner />
        let orderSummery=null;
        //console.log('ingr',this.props.ings)
        if(this.props.ings){
            burger=(
                <Aux>
                    <Burger ingredients={this.props.ings} />
                    <BuildControls orderedNow={this.orderNowhandler} purchasable={this.updatePurchable(this.props.ings)} price={this.props.price} disabledInfo={disabledInfo}  addedIngredients={this.props.onIngredientAdded} removeIngredients={this.props.onIngredientRemoved} />
                </Aux>
            )
            orderSummery= <OrderSummery totalPrice={this.props.price} cancelHander={this.ModalClosedHandler} continueHander={this.ContinueOrderHandler} ingredients={this.props.ings}></OrderSummery>
        }
        if(this.props.error){
            burger=<p>ingredients are not loaded...</p>
        }
        if(this.state.loading){
            orderSummery =<Spinner />
        }
        return(
            <Aux>
                <Modal show={this.state.purchasing} modalClosed={this.ModalClosedHandler}>
                    {orderSummery}
                </Modal>
                {burger}
            </Aux>
        );
    }
}

const mapStateToProps = state =>{
    return {
        ings :state.burgerBuilder.ingredients,
        price:state.burgerBuilder.totalPrice,
        error:state.burgerBuilder.error
    };
}
const mapDispatchToProps = dispatch =>{
    return {
        onIngredientAdded:(ingrName)=>dispatch(addIngredients(ingrName)),
        onIngredientRemoved:(ingrName)=>dispatch(removeIngredients(ingrName)),
        loadIngredients:()=>dispatch(initIngredients())
    };
}
export default connect(mapStateToProps,mapDispatchToProps)(WithErrorHandler(BurgerBuilder,axios));