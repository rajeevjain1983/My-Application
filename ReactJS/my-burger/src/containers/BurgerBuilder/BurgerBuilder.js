import React, {Component} from 'react';
import Aux from '../../hoc/Aux/Aux';
import Burger from '../../components/Burger/Burger';
import BuildControls from '../../components/Burger/BuildControls/BuildControls';
import Modal from '../../components/UI/Modal/Modal';
import OrderSummery from '../../components/Burger/OrderSummery/OrderSummery';
import axios from '../../axios';
import Spinner from '../../components/UI/Spinner/Spinner';
import WithErrorHandler from '../../hoc/WithErrorHandler/WithErrorHander';

const INGREDIENTS_PRICE={
    salad:5,
    bacon:6,
    cheese:7,
    meat:10
};
class BurgerBuilder extends Component{

    
    state={
        ingredients:null,
        totalPrice:10,
        purchasable:false,
        purchasing:false,
        loading:false,
        error:false
    }

    orderNowhandler =()=>{
        this.setState({purchasing:true});
    }
    updatePurchable=(ingredients)=>{
        const sum= Object.keys(ingredients).map(igkey=>{
            return ingredients[igkey];
        })
        .reduce((sum,el)=>{
            return sum+el;
        },0)
        
        this.setState({purchasable:sum>0})
    }
    addIngredientsHandler=(type)=>{
        const oldCount= this.state.ingredients[type];
        const updatedCount= oldCount+1;
        const updatedIngredients={...this.state.ingredients};
        updatedIngredients[type]=updatedCount;
        const oldPrice=this.state.totalPrice;
        const updatedPrice=oldPrice+INGREDIENTS_PRICE[type];
        this.setState({totalPrice:updatedPrice,ingredients:updatedIngredients})
        this.updatePurchable(updatedIngredients);
    }
    removeIngredientsHandler=(type)=>{
        const oldCount= this.state.ingredients[type];

        if(oldCount<=0)
            return;
        const updatedCount= oldCount-1;
        const updatedIngredients={...this.state.ingredients};
        updatedIngredients[type]=updatedCount;
        const oldPrice=this.state.totalPrice;
        const updatedPrice=oldPrice-INGREDIENTS_PRICE[type];
        this.setState({totalPrice:updatedPrice,ingredients:updatedIngredients})
        this.updatePurchable(updatedIngredients);
    }
    ModalClosedHandler = ()=>{
        this.setState({purchasing:false});
    }
    ContinueOrderHandler=()=>{
        const quryParams =[];
        for(let i in this.state.ingredients){
            quryParams.push(encodeURIComponent(i) + '='+ encodeURIComponent(this.state.ingredients[i]));
        }
        quryParams.push('price='+this.state.totalPrice);
        const queryString = quryParams.join('&');
        // console.log(queryString);
       this.props.history.push({
           pathname:'/checkout',
           search: '?'+queryString
        });
        
    }
    componentDidMount(){
        // console.log(this.props);
        axios.get('/ingredients.json')
        .then(res=>{
            console.log(res.data);
            this.setState({ingredients:res.data})
        })
        .catch(error=>{
            console.log(error);
            this.setState({error:true});
        })
    }
    render(){
        const disabledInfo={...this.state.ingredients};
        for(let key in disabledInfo){
            disabledInfo[key]=disabledInfo[key]<=0;
        }

        let burger=<Spinner />
        let orderSummery=null;
        if(this.state.ingredients){
            burger=(
                <Aux>
                    <Burger ingredients={this.state.ingredients} />
                    <BuildControls orderedNow={this.orderNowhandler} purchasable={this.state.purchasable} price={this.state.totalPrice} disabledInfo={disabledInfo}  addedIngredients={this.addIngredientsHandler} removeIngredients={this.removeIngredientsHandler} />
                </Aux>
            )
            orderSummery= <OrderSummery totalPrice={this.state.totalPrice} cancelHander={this.ModalClosedHandler} continueHander={this.ContinueOrderHandler} ingredients={this.state.ingredients}></OrderSummery>
        }
        if(this.state.error){
            burger=<p>ingredients are not loaded...</p>
        }
        if(this.state.loading){
            orderSummery =<Spinner />
        }
        return(
            <Aux>
                <Modal show={this.state.purchasing} modalClosed={this.ModalClosedHandler}>
                    {orderSummery}
                </Modal>
                {burger}
            </Aux>
        );
    }
}
export default WithErrorHandler(BurgerBuilder,axios);