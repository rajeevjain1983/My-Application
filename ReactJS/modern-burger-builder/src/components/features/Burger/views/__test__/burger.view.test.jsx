import React from "react";
import Adapter from "enzyme-adapter-react-16";
import { shallow, configure } from "enzyme";
import BurgerView from "../burger.view";
import Modal from "../../../../common/UI/Modal";
import BurgerBuilder from "../../organism/BurgerBuilder";
import BurgerBuildControls from "../../organism/BurgerBuildControls";

configure({ adapter: new Adapter() });

describe("<BurgerView />", () => {
  const props = {
    ingredients: {
      Meat: 1,
      Bacon: 1,
      Cheese: 1,
    },
    ingredientsHandler: () => {},
    burgerPrice: 23,
    purchasable: true,
    show: true,
    showModal: () => {},
    hideModal: () => {},
  };
  it("Should match snapshot", () => {
    const wrapper = shallow(<BurgerView {...props} />);
    console.log(wrapper);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find(Modal)).toHaveLength(1);
    expect(wrapper.find(BurgerBuilder)).toHaveLength(1);
    expect(wrapper.find(BurgerBuildControls)).toHaveLength(1);
  });
});
