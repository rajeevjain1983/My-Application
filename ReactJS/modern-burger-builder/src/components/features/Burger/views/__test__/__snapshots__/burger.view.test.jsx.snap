// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BurgerView /> Should match snapshot 1`] = `
<Fragment>
  <withStyle
    show={true}
    showModal={[Function]}
  >
    <withStyle
      burgerPrice="23"
      ingredients={
        Object {
          "Bacon": 1,
          "Cheese": 1,
          "Meat": 1,
        }
      }
      showModal={[Function]}
    />
  </withStyle>
  <withStyle
    ingredients={
      Object {
        "Bacon": 1,
        "Cheese": 1,
        "Meat": 1,
      }
    }
  />
  <burgerview__PriceLabel>
    Burger Price : $
    23
  </burgerview__PriceLabel>
  <withStyle
    ingredients={
      Object {
        "Bacon": 1,
        "Cheese": 1,
        "Meat": 1,
      }
    }
    ingredientsHandler={[Function]}
    purchasable={true}
    showModal={[Function]}
  />
</Fragment>
`;
